claims.go

```diff
@@ -9,7 +9,7 @@ import (
 // Claims must just have a Valid method that determines
 // if the token is invalid for any supported reason
 type Claims interface {
-	Valid() error
+	Valid(opts ...validationOption) error
```

Comment:
I think we should add a comment, either on this interface (or maybe even better) around the `validationOption` explaining why it is not an exported identifier. 

Looking at this method signature I'd be a little confused.

Reply:
Since the interface changed to include an unexported type, it is impossible to implement user-defined `Claims` (and `Token` since it contains `Claims`)?

Reply:
You should be able to implement user-defined `Claims` by embedding either `RegisteredClaims` or the old `StandardClaims`. It could be impossible to completely implement a different claims structure that does not re-use `RegisteredClaims`, but that is not intended anyway.

See our example here https://github.com/golang-jwt/jwt/blob/d489c99d3e8a82bd47dcb3e4880414f7f747eaa7/example_test.go#L33-L39

Reply:
This breaks backwards compatibility for anyone using older `v3.x.y` and `dgrijalva/jwt-go`:

```
cannot use &"github.com/golang-jwt/jwt/v4".MapClaims{} (type *"github.com/golang-jwt/jwt/v4".MapClaims) as type "github.com/golang-jwt/jwt".Claims in field value:
	*"github.com/golang-jwt/jwt/v4".MapClaims does not implement "github.com/golang-jwt/jwt".Claims (wrong type for Valid method)
		have Valid(..."github.com/golang-jwt/jwt/v4".validationOption) error
		want Valid() error
```
Can you please consider adding a new function (`ValidWithOptions`) instead of changing `Valid()` funtion signature?